spec/models/finance/schedule/npq_leadership_spec.rb

```diff
@@ -13,10 +13,136 @@
+
+    it "returns NPQ Leadership Autumn 2022 schedule" do
+      expected_schedule = described_class.find_by(cohort:, schedule_identifier: "npq-leadership-spring")
+
+      expect(described_class.default_for(cohort:)).to eql(expected_schedule)
+    end
+  end
+
+  describe ".schedule_for" do
+    let(:cohort)            { Cohort.find_by!(start_year: 2022) }
+    let(:cohort_start_year) { cohort.start_year }
+
+    context "when date is between June and December of cohort start year" do
+      it "returns NPQ Leadership Autumn schedule" do
+        expected_schedule = described_class.find_by(cohort:, schedule_identifier: "npq-leadership-autumn")
+
+        travel_to Date.new(cohort_start_year, 6, 1) do
+          expect(described_class.schedule_for(cohort:)).to eq(expected_schedule)
+        end
+      end
+    end
+
+    context "when date is between December of cohort start year and April of the next year" do
+      it "returns NPQ Leadership Spring schedule" do
+        expected_schedule = described_class.find_by(cohort:, schedule_identifier: "npq-leadership-spring")
+
+        travel_to Date.new(cohort_start_year, 12, 26) do
+          expect(described_class.schedule_for(cohort:)).to eq(expected_schedule)
+        end
+      end
+    end
+
+    context "when date is between April and December of the next year" do
+      it "returns NPQ Leadership Autumn schedule" do
+        expected_schedule = described_class.find_by(cohort:, schedule_identifier: "npq-leadership-autumn")
+
+        travel_to Date.new(cohort_start_year + 1, 4, 16) do
+          expect(described_class.schedule_for(cohort:)).to eq(expected_schedule)
+        end
+      end
+    end
+
+    context "when date is between December of next year and April in 2 years" do
+      it "returns NPQ Leadership Spring schedule" do
+        expected_schedule = described_class.find_by(cohort:, schedule_identifier: "npq-leadership-spring")
+
+        travel_to Date.new(cohort_start_year + 1, 12, 26) do
+          expect(described_class.schedule_for(cohort:)).to eq(expected_schedule)
+        end
+      end
+    end
+
+    context "when date range exceeds the current cohort" do
+      it "returns default schedule for cohort" do
+        expected_schedule = described_class.find_by(cohort:, schedule_identifier: "npq-leadership-autumn")
+
+        travel_to Date.new(cohort_start_year + 1, 10, 1) do
+          expect(described_class.schedule_for(cohort:)).to eq(expected_schedule)
+        end
+      end
+    end
+
+    context "when no schedule exists for the cohort" do
+      let(:cohort) { create(:cohort, start_year: 2020) }
+
+      it "raises an error" do
+        expect { described_class.schedule_for(cohort:) }.to raise_error(ActiveRecord::RecordNotFound)
+      end
+    end
+  end
+
+  describe ".spring_schedule?" do
+    it "returns true when date between Dec 26 to Apr 15" do
+      (("2021-12-26".to_date)..("2022-04-15".to_date)).each do |date|
```

Comment:
is it possible to avoid hardcoding the year, and using relative years across those specs 🤔